projects/feature-activation/0002-bit-signaling.md

```diff
@@ -0,0 +1,64 @@
+
+Given the [Feature Activation for Blocks RFC](./0001-feature-activation-for-blocks.md), a mechanism is provided to deploy new features in the Hathor network conditionally on some criteria. Part of the defined criteria is the ability for miners to signal their support for a new feature by setting signal bits on a mined block. The former RFC describes how this bit is calculated, interpreted, etc., but details on how miners would actually set those bits were left open. This RFC addresses this requirement by defining a user-friendly way for miners to configure their support for a feature, or the lack thereof.
+
+# Motivation
+[motivation]: #motivation
+
+In the current implementation, the Feature Activation process can be used but would require manual manipulation of block bits by miners to set support for features undergoing the process. Instead, we should provide a user-friendly way through our full node CLI interface so miners can easily start their mining full node with the desired feature support configuration. That configuration will then be reflected on mining block templates, so the templates carry the corresponding signal bits for all features.
+
+# Guide-level explanation
+[Guide-level explanation]: #guide-level-explanation
+
+Users running the full node, or specifically miners in this case, should be able to pass two new options to the `run_node` CLI interface:
+
+1. `--signal-support [feature]`, and
+2. `--signal-not-support [feature]`
+
+Here are rule definitions for these options:
+
+1. Both options receive the name of some feature, as defined in the existing `Feature` enum;
+2. Both options can be used multiple times to either enable or disable support for multiple features;
+3. A feature cannot be set in both options at the same time, as this would be a conflicting configuration;
+4. A feature can only be set in any of these options if it's currently undergoing a signaling phase of its Feature Activation process, otherwise setting its bit could conflict with other features;
+5. A default bit signal value (either enabled or disabled) should be provided for each feature.
+
+Here's an example of running the full node with those options, enabling support for features 2 and 3, and disabling support for feature 1:
+
+```bash
+poetry run hathor-cli run_node --status 8080 --testnet --memory-storage --signal-not-support NOP_FEATURE_1 --signal-support NOP_FEATURE_2 --signal-support NOP_FEATURE_3
+```
+
+Let's assume all features are disabled by default, and that `NOP_FEATURE_1` is configured to use `bit=0`, `NOP_FEATURE_2` is configured to use `bit=1`, and `NOP_FEATURE_3` is configured to use `bit=2`, where `bit=0` is the LSB, as explained in the previous RFC.
+
+Then, the signal bits for this full node run would be `110`, meaning that those bits would be present in all block templates generated for mining.
+
+# Reference-level explanation
+[Reference-level explanation]: #reference-level-explanation
+
+## Bit Signaling Serivce
```

Comment:
typo

Reply:
Thanks! Fixed in https://github.com/HathorNetwork/rfcs/pull/59/commits/c0ec7e86771268e26cb2adf63bc51a31942fd3f1

Reply:
Ok.